### PR TITLE
refactor: Address MiniMessage deprecations

### PR DESCRIPTION
**Pull request by NotMyFault**
## Description
According to OSS Sonatype, 4.1.0-SNAPSHOT was last updated back in July 2021, either renovate missed it or there was no release of 4.2.0. However, there are over 600 deprecation & "scheduled for removal" warnings we should address in a timely manner.

## Checklist

- [X] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [X] Ensure that the pull request title represents the desired changelog entry.
- [X] I tested my changes and approved their functionality.
- [X] I ensured my changes do not break other parts of the code
- [X] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Comment by NotMyFault** [comment]
Note (to myself): Update HoloPlots' after this is pulled, we are using a single Template there too.

File: Core/src/main/java/com/plotsquared/core/command/FlagCommand.java [hunk 1]
```diff
@@ -576,20 +577,22 @@ public void list(
         for (final Map.Entry<String, ArrayList<String>> entry : flags.entrySet()) {
             Collections.sort(entry.getValue());
             Component category =
-                    MINI_MESSAGE.parse(
+                    MINI_MESSAGE.deserialize(
                             TranslatableCaption.of("flag.flag_list_categories").getComponent(player),
-                            Template.of("category", entry.getKey())
+                            TemplateResolver.templates(Template.template("category", entry.getKey()))
```
Comment (dordsor21) [inline Core/src/main/java/com/plotsquared/core/command/FlagCommand.java:582]: Is TemplateResolver required here?

Comment (NotMyFault) [reply Core/src/main/java/com/plotsquared/core/command/FlagCommand.java:582]: Yes, #deserialize expects a TemplateResolver.

File: Core/src/main/java/com/plotsquared/core/player/PlotPlayer.java [hunk 2]
```diff
@@ -866,9 +867,9 @@ public void sendTitle(
             final int fadeIn, final int stay, final int fadeOut,
             final @NonNull Template... replacements
     ) {
-        final Component titleComponent = MiniMessage.get().parse(title.getComponent(this), replacements);
+        final Component titleComponent = MiniMessage.miniMessage().deserialize(title.getComponent(this), TemplateResolver.templates(replacements));
         final Component subtitleComponent =
-                MiniMessage.get().parse(subtitle.getComponent(this), replacements);
+                MiniMessage.miniMessage().deserialize(subtitle.getComponent(this), TemplateResolver.templates(replacements));
```
Comment (dordsor21) [inline Core/src/main/java/com/plotsquared/core/player/PlotPlayer.java:872]: Should we not be statically storing a MiniMessage here?

Comment (NotMyFault) [reply Core/src/main/java/com/plotsquared/core/player/PlotPlayer.java:872]: 4.1.0 -> 4.2.0 deprecates naming schemes e.g. the getter in MiniMessage and replaces it with #miniMessage. The TemplateResolver was already used under the hood in #parse, same like for #miniMessage, it does the same like #get. This is a chore PR that addresses the deprecation scheme, I didn't modify public API, more changes to MM is coming soon tm.

File: Core/src/main/java/com/plotsquared/core/util/placeholders/PlaceholderRegistry.java [hunk 3]
```diff
@@ -80,7 +80,7 @@ public PlaceholderRegistry(final @NonNull EventDispatcher eventDispatcher) {
      * @return a legacy-formatted string.
      */
     private static String legacyComponent(TranslatableCaption caption, LocaleHolder localeHolder) {
-        Component component = MiniMessage.get().parse(caption.getComponent(localeHolder));
+        Component component = MiniMessage.miniMessage().parse(caption.getComponent(localeHolder));
```
Comment (dordsor21) [inline Core/src/main/java/com/plotsquared/core/util/placeholders/PlaceholderRegistry.java:83]: Staticate?

File: Core/src/main/java/com/plotsquared/core/player/PlotPlayer.java (hunk 2 above)
Comment (dordsor21) [reply Core/src/main/java/com/plotsquared/core/player/PlotPlayer.java:872]: ? I mean adding a `private static final MiniMessage MINI_MESSAGE = MiniMessage.miniMessage();`

Comment (NotMyFault) [reply Core/src/main/java/com/plotsquared/core/player/PlotPlayer.java]: Yeah, that was solely for clarification.

**Review by dordsor21 — APPROVED** [review]

**Comment by NotMyFault** [comment]
I would like to pull this tomorrow so I can update HoloPlots too if I'm at work, if there's no negative feedback.

**Pull request merged**